PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java

```diff
@@ -379,18 +380,22 @@ public void openLinkInBrowser(@NotNull String url) {
     @SneakyThrows
     public void openAppServiceFile(final AppServiceFile target, Object context) {
         final AppServiceFileService fileService = AppServiceFileService.forApp(target.getApp());
-        final AppServiceFile file = fileService.getFileByPath(target.getPath());
-        if (file == null) {
-            UIUtil.invokeLaterIfNeeded(() -> Messages.showWarningDialog(String.format("Target file %s has been deleted", target.getName()), "Open File"));
-            return;
-        }
         final FileEditorManager fileEditorManager = FileEditorManager.getInstance((Project) context);
-        final VirtualFile virtualFile = getOrCreateVirtualFile(file, fileEditorManager);
+        final VirtualFile virtualFile = getOrCreateVirtualFile(target, fileEditorManager);
         final OutputStream output = virtualFile.getOutputStream(null);
         final String failure = String.format("Can not open file %s. Try downloading it first and open it manually.", virtualFile.getName());
+        final String failureFileDeleted = String.format("Target file %s has been deleted", target.getName());
         final String title = String.format("Opening file %s...", virtualFile.getName());
-        final AzureTask task = new AzureTask(null, title, true, () -> {
-            ProgressManager.getInstance().getProgressIndicator().setIndeterminate(true);
+        final AzureTask<Void> task = new AzureTask<>(null, title, true, () -> {
+            final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+            indicator.setIndeterminate(true);
+            indicator.setText2("Checking file existence");
+            final AppServiceFile file = fileService.getFileByPath(target.getPath());
+            if (file == null) {
+                UIUtil.invokeLaterIfNeeded(() -> Messages.showWarningDialog(failureFileDeleted, "Open File"));
```

Comment:
move the stat ` final String failureFileDeleted = String.format("Target file %s has been deleted", target.getName());` to here